backend/app/services/scheduled_inquiries.py

```diff
@@ -1,32 +1,103 @@
-from sqlmodel import Session, col, desc, func, select
+import logging
+
+from sqlmodel import Session, col, desc, func, select, update
 
 from app.models import (
     Inquiry,
     ScheduledInquiry,
     ScheduledInquiryBase,
     ScheduledInquiryPublic,
+    ScheduledInquiryUpdate,
 )
 from app.models.scheduled_inquiry import ScheduledInquiryPublicWithInquiryText
 
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
 
 def create(*, session: Session, inquiry_id: int) -> ScheduledInquiryPublic:
     highest_rank = session.exec(
         select(ScheduledInquiry.rank).order_by(desc(ScheduledInquiry.rank))
     ).first()
-
     rank = highest_rank + 1 if highest_rank else 1
-
     scheduled_inquiry = ScheduledInquiryBase(inquiry_id=inquiry_id, rank=rank)
-
     db_inquiry = ScheduledInquiry.model_validate(scheduled_inquiry)
-
     session.add(db_inquiry)
     session.commit()
     session.refresh(db_inquiry)
-
     return ScheduledInquiryPublic.model_validate(db_inquiry)
 
 
+def enable_scheduled_inquiry(
+    *, session: Session, scheduled_inquiry_id: int
+) -> ScheduledInquiryPublic:
+    stmt = select(ScheduledInquiry).where(ScheduledInquiry.id == scheduled_inquiry_id)
+    target: ScheduledInquiry | None = session.exec(stmt).first()
+    if not target:
+        raise ValueError("Invalid scheduled_inquiry_id to enable")
+    highest_rank = session.exec(
+        select(ScheduledInquiry.rank).order_by(desc(ScheduledInquiry.rank))
+    ).first()
+    try:
+        target.sqlmodel_update({"rank": highest_rank + 1 if highest_rank else 1})
+        session.commit()
+    except Exception:
+        raise ValueError("Invalid scheduled inquiry update request")
+    session.refresh(target)
+    return ScheduledInquiryPublic.model_validate(target)
+
+
+def disable_scheduled_inquiry(
+    *, session: Session, scheduled_inquiry_id: int
+) -> ScheduledInquiryPublic:
+    stmt = select(ScheduledInquiry).where(ScheduledInquiry.id == scheduled_inquiry_id)
+    target: ScheduledInquiry | None = session.exec(stmt).first()
+    if not target:
+        raise ValueError("Invalid scheduled_inquiry_id to disable")
+    try:
+        target.sqlmodel_update({"rank": 0})
+        session.commit()
+    except Exception:
+        raise ValueError("Invalid scheduled inquiry update request")
+    session.refresh(target)
+    return ScheduledInquiryPublic.model_validate(target)
+
+
+def update_scheduled_inquiry(
+    *, session: Session, scheduled_inquiry_in: ScheduledInquiryUpdate
+) -> ScheduledInquiryPublic:
+    stmt = select(ScheduledInquiry).where(
```

Comment:
nit -- getting the ScheduledInquiry by ID is around 5 lines of code consistently repeated. Maybe consider refactor into a function?

Reply:
Sure, I will make this adjustment.